### PR TITLE
chore: upgrade to typescript 3.0.x

### PR DESCRIPTION
**Pull request by raymondfeng**
The PR allows typedoc to be used with typescript 3.0.x. Once the PR is merged, it should be released as new minor version.

### CODE REVIEW

File: src/lib/ts-internal.ts [hunk 1]
```diff
@@ -98,7 +98,7 @@ export function isBindingPattern(node: ts.Node): node is ts.BindingPattern {
 
 // https://github.com/Microsoft/TypeScript/blob/v2.1.4/src/compiler/utilities.ts#L1729
 export function getClassExtendsHeritageClauseElement(node: ts.ClassLikeDeclaration | ts.InterfaceDeclaration) {
-  return tsany.getClassExtendsHeritageClauseElement.apply(this, arguments);
+  return tsany.getClassExtendsHeritageElement.apply(this, arguments);
```
Comment (aciccarello) [inline src/lib/ts-internal.ts]: This should probably be a separate PR and the function name should be updated

Comment (raymondfeng) [reply src/lib/ts-internal.ts]: +1 to rename the exported function in a separate PR.

Comment (raymondfeng) [reply src/lib/ts-internal.ts]: @aciccarello Do you think it's better to keep both old and new exported functions for backward compatibility?

Comment (aciccarello) [reply src/lib/ts-internal.ts]: Since it is only used internally, I'd say change the calling function (I think there is only one) and get rid of any `getClassExtendsHeritageClauseElement` references.

Comment (aciccarello) [reply src/lib/ts-internal.ts]: This gave me an idea for #830 which is related
